SendGrid/SendGrid/Client.cs

```diff
@@ -0,0 +1,125 @@
+        public APIKeys ApiKeys;
+        public string Version;
+        private Uri _baseUri;
+        private const string MediaType = "application/json";
+        private enum Methods
+        {
+            GET, POST, PATCH, DELETE
+        }
+
+        /// <summary>
+        ///     Create a client that connects to the SendGrid Web API
+        /// </summary>
+        /// <param name="apiKey">Your SendGrid API Key</param>
+        /// <param name="baseUri">Base SendGrid API Uri</param>
+        public Client(string apiKey, string baseUri = "https://api.sendgrid.com/")
+        {
+            _baseUri = new Uri(baseUri);
+            _apiKey = apiKey;
+            Version = Assembly.GetExecutingAssembly().GetName().Version.ToString();
+            ApiKeys = new APIKeys(this);
+        }
+
+        /// <summary>
+        ///     Create a client that connects to the SendGrid Web API
+        /// </summary>
+        /// <param name="method">HTTP verb, case-insensitive</param>
+        /// <param name="endpoint">Resource endpoint, do not prepend slash</param>
+        /// <param name="data">An JObject representing the resource's data</param>
+        /// <returns>An asyncronous task</returns>
+        private async Task<HttpResponseMessage> RequestAsync(Methods method, string endpoint, JObject data)
+        {
+            using (var client = new HttpClient())
+            {
+                try
+                {
+                    client.BaseAddress = _baseUri;
+                    client.DefaultRequestHeaders.Accept.Clear();
+                    client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue(MediaType));
+                    client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", _apiKey);
+                    client.DefaultRequestHeaders.TryAddWithoutValidation("User-Agent", "sendgrid/" + Version + ";csharp");
+
+                    switch (method)
+                    {
+                        case Methods.GET:
+                            return await client.GetAsync(endpoint);
+                        case Methods.POST:
+                            return await client.PostAsJsonAsync(endpoint, data);
+                        case Methods.PATCH:
+                            endpoint = _baseUri + endpoint;
+                            StringContent content = new StringContent(data.ToString(), Encoding.UTF8, MediaType);
+                            HttpRequestMessage request = new HttpRequestMessage
+                            {
+                                Method = new HttpMethod("PATCH"),
+                                RequestUri = new Uri(endpoint),
+                                Content = content
+                            };
+                            return await client.SendAsync(request);
+                        case Methods.DELETE:
+                            return await client.DeleteAsync(endpoint);
+                        default:
```

Comment:
Now that we have enumerations...potentially the default case would never happen right?